app/models/grda_warehouse/warehouse_reports/project/data_quality/version_three.rb

```diff
@@ -0,0 +1,1988 @@
+          support["bed_utilization_#{project.id}_#{attr}"] = {
+            headers: ['Client ID', 'First Name', 'Last Name'],
+            counts: data[attr]
+          }
+        end
+
+        bed_utilization << project_counts
+
+      end
+      self.class.bed_utilization_attributes.each do |attr|
+        totals[:counts]["#{attr}_percentage"] = in_percentage(totals[:counts][attr], totals[:counts][:capacity])
+        support["bed_utilization_totals_#{attr}"] = {
+          headers: ['Client ID', 'First Name', 'Last Name'],
+          counts: totals[:data][attr]
+        }
+      end
+
+      add_answers(
+        {
+          bed_utilization: bed_utilization,
+          bed_utilization_totals: totals,
+        },
+        support
+      )
+
+    end
+
+    def self.bed_utilization_attributes
+      [
+        :average_daily,
+        :first_of_month,
+        :fifteenth_of_month,
+        :last_of_month,
+      ]
+    end
+
+    def set_bed_coverage_data
+      bed_coverage = 0
+      bed_coverage_percent = 0
+      if hmis_beds > 0
+        bed_coverage = "#{beds} / #{hmis_beds}" rescue 0
+        bed_coverage_percent = (beds.to_f/hmis_beds*100).round(2) || 0
+      end
+      add_answers({
+        bed_coverage: bed_coverage,
+        bed_coverage_percent: bed_coverage_percent,
+      })
+    end
+
+    def add_unit_utilization
+      unit_utilization = []
+      support = {}
+      totals = {counts: Hash.new(0), data: Hash.new(Set.new)}
+
+      client_columns = [:client_id, c_t[:FirstName].as('first_name').to_sql, c_t[:LastName].as('last_name').to_sql]
+      filter = ::Filters::DateRange.new(start: self.start, end: self.end)
+      projects.each do |project|
+        counts = {}
+        data = {}
+
+        counts[:capacity] = project.inventories.within_range(filter.range).map{|i| i[:UnitInventory] || 0}.sum
+
+        household_ids = households
+
+        total_count = household_ids.count
+        counts[:average_daily] = total_count / filter.range.count rescue 0
+
+        data[:average_daily] = project.service_history_enrollments.
+            joins(:client, :enrollment).
+            where(Enrollment: {RelationshipToHoH: 1}).
+            where(date: filter.range).
+            distinct.
+            pluck(*client_columns)
+
+        filter.range.each do |date|
+          key = date.to_formatted_s(:iso8601)
+          data[key] = project.service_history_enrollments.
+              joins(:client, :enrollment).
+              where(Enrollment: {RelationshipToHoH: 1}).
+              where(date: date).
+              distinct.
+              pluck(*client_columns)
+          counts[key] = data[key].count
+        end
+
+        project_counts = {
+            id: project.id,
+            name: project.name,
+            project_type: project[GrdaWarehouse::Hud::Project.project_type_column],
+        }.merge(counts)
+
+        totals[:counts][:capacity] += counts[:capacity]
+        [:average_daily].each do |attr|
```

Comment:
continuing the conversation from above about data[:average_daily] this doesn't look correct `[:average_daily].each`  in `VersionTwo` this is something like `self.class.bed_utilization_attributes.each do |attr|`